src/components/pagination/pager-pagination.tsx

```diff
@@ -0,0 +1,88 @@
+import React, {Fragment, useEffect, useState} from "react";
+import {Pagination} from "react-bootstrap";
+import {PagerItem} from "./pager-item";
+
+export interface PaginationProps {
+    numberOfPageButtonsToShowAfterAndBeforeCurrent: number
+    onPageChange: (pageIndex: number) => void
+    lastPageIndex: number
+}
+
+export const PagerPagination: React.FC<PaginationProps> = ({numberOfPageButtonsToShowAfterAndBeforeCurrent, onPageChange, lastPageIndex}) => {
+    if (numberOfPageButtonsToShowAfterAndBeforeCurrent % 2 !== 0) {
+        throw new Error("number of pages to show must be even!")
+    }
+
+    const [pageIndex, setPageIndex] = useState(0);
+    const correctedPageIndex = Math.min(pageIndex, lastPageIndex);
+    const wantedUpperPageIndex = correctedPageIndex + numberOfPageButtonsToShowAfterAndBeforeCurrent;
+    const wantedLowerPageIndex = correctedPageIndex - numberOfPageButtonsToShowAfterAndBeforeCurrent;
+
+    useEffect(() => {
+        onPageChange(pageIndex)
+    }, [onPageChange, pageIndex])
+
+    useEffect(() => {
+        if (pageIndex !== correctedPageIndex) {
+            setPageIndex(correctedPageIndex);
+        }
+    }, [pageIndex, correctedPageIndex]);
```

Comment:
maybe you want to the replace this with a function wrapping setPageIndex with this check (and the correctedPageIndex calculation)

Reply:
This function is no effect anymore.

Reply:
Why?

Reply:
useEffect is excuted after the rendering, but I need the correctedPageIndex before the rendering.

Reply:
because with this you change the value, then a new calculation of `correctedPageIndex` is triggered and then this useEffect kicks in and possibly changes the value again.
With a wrapper function the change would be happen once

Reply:
If you think it's better then leave the calculation of `correctedPageIndex` out of it. But the check could go in such a wrapper function

Reply:
I removed the effect because it doesn't change the behaviour in a useful way.